ai_eval/base.py

```diff
@@ -0,0 +1,138 @@
+"""Base Xblock with AI evaluation."""
+
+import pkg_resources
+
+from django.utils.translation import gettext_noop as _
+from xblock.core import XBlock
+from xblock.fields import String, Scope, Dict
+from xblock.validation import ValidationMessage
+
+
+from .llm import SupportedModels
+
+try:
+    from xblock.utils.studio_editable import StudioEditableXBlockMixin
+except ModuleNotFoundError:  # For compatibility with Palm and earlier
+    from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+try:
+    from xblock.utils.resources import ResourceLoader
+except (
+    ModuleNotFoundError
+):  # For backward compatibility with releases older than Quince.
+    from xblockutils.resources import ResourceLoader
+
+
+class AIEvalXBlock(StudioEditableXBlockMixin, XBlock):
+    """
+    Base class for Xblocks with AI evaluation
+    """
+
+    USER_KEY = "USER"
+    LLM_KEY = "LLM"
+
+    loader = ResourceLoader(__name__)
+
+    icon_class = "problem"
+    model_api_key = String(
+        display_name=_("Chosen model API Key"),
+        help=_("Enter your the API Key of your chosen model."),
+        default="",
+        scope=Scope.settings,
+    )
+    model_api_url = String(
+        display_name=_("Set your API URL"),
+        help=_(
+            "This required with models that don't have an official provider such as Llama."
```

Comment:
@CefBoud Nit - it might help to provide an example url in the help text for this field.